apisix/discovery/nacos.lua

```diff
@@ -0,0 +1,316 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local require            = require
+local local_conf         = require("apisix.core.config_local").local_conf()
+local http               = require("resty.http")
+local core               = require("apisix.core")
+local ipairs             = ipairs
+local type               = type
+local math               = math
+local math_random        = math.random
+local error              = error
+local ngx                = ngx
+local ngx_re             = require("ngx.re")
+local ngx_timer_at       = ngx.timer.at
+local ngx_timer_every    = ngx.timer.every
+local string             = string
+local string_sub         = string.sub
+local str_byte           = string.byte
+local str_find           = core.string.find
+local log                = core.log
+
+local default_weight
+local applications
+local auth_path = "auth/login"
+local instance_list_path = "ns/instance/list?healthyOnly=true&serviceName="
+
+local schema = {
+    type = "object",
+    properties = {
+        host = {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "string",
+            },
+        },
+        fetch_interval = {type = "integer", minimum = 1, default = 30},
+        prefix = {type = "string", default = "/nacos/v1/"},
+        weight = {type = "integer", minimum = 1, default = 100},
+        timeout = {
+            type = "object",
+            properties = {
+                connect = {type = "integer", minimum = 1, default = 2000},
+                send = {type = "integer", minimum = 1, default = 2000},
+                read = {type = "integer", minimum = 1, default = 5000},
+            },
+            default = {
+                connect = 2000,
+                send = 2000,
+                read = 5000,
+            }
+        },
+    },
+    required = {"host"}
+}
+
+
+local _M = {
+    version = 0.1,
+}
+
+
+local function request(request_uri, path, body, method, basic_auth)
+    local url = request_uri .. path
+    log.info("request url:", url)
+    local headers = core.table.new(0, 0)
+    headers['Accept'] = 'application/json'
```

Comment:
for the string object, I think we use `"`, it is easier for reading.

`headers["Accept"] = "application/json"`

Reply:
and please fix the similar points